src/usermod.c

```diff
@@ -791,9 +791,9 @@ static void update_group (void)
 			SYSLOG ((LOG_WARN, "failed to prepare the new %s entry '%s'", gr_dbname (), ngrp->gr_name));
 			fail_exit (E_GRP_UPDATE);
 		}
-
-		gr_free(ngrp);
 	}
+
+	gr_free(ngrp);
 }
 
```

Comment:
This seems to have the same issue as in the other patch for this file.

Reply:
Right, so I offered a similar solution.